### PR TITLE
Fix efficiency bug in MemMappedDynamicDirectedGraphConverter. 

### PR DESCRIPTION
**Pull request by peterl-stripe**
And for simplicity, drop support for graphs with between 2**31 and 2**32 edges for now.

@gkk-stripe could you review this?  This is a bug @atausz-stripe found in Tempest, where if the edge ids happen to be sorted (and the id range for targetIds is much smaller than the id range for sourceIds), the BigArrayLists in MemMappedDynamicDirectedGraphConverter which store degrees grow one entry at a time for O(n^2) total time (instead of doubling capacity as needed, for O(n) total time). I think this was introduced when I tried to add support for for than 2**31 nodes, but we don't currently need that, so for now I dropped support for converting such graphs.  We can add support for that back as needed.

CC @teapot-co/tempest

### CODE REVIEW

File: src/main/scala/co/teapot/tempest/graph/MemMappedDynamicDirectedGraphConverter.scala [hunk 1]
```diff
@@ -42,33 +42,33 @@ object MemMappedDynamicDirectedGraphConverter {
...
+    val outDegrees = new IntArrayList()
+    val inDegrees = new IntArrayList()
 
     Util.logWithRunningTime(log, "first pass", printAtStart = true) {
-      FileUtil.forEachLongPair(edgeListFile, log, linesPerMessage = 1000000) { (id1, id2) =>
-        if (id1 >= outDegrees.size64) {
-          outDegrees.size(id1 + 1) // Note that size fills new entries with 0s
+      FileUtil.forEachIntPair(edgeListFile, log, linesPerMessage = 1000000) { (id1, id2) =>
+        if (id1 >= outDegrees.size) {
+          outDegrees.addAll(IntArrayList.wrap(new Array[Int](id1 + 1 - outDegrees.size)))
```
Comment (gkk-stripe) [inline src/main/scala/co/teapot/tempest/graph/MemMappedDynamicDirectedGraphConverter.scala]: I'm not sure I follow here. When `id1 = outDegrees.size` you get `outDegrees.size + 1 - outDegrees.size = 1` so it seems like you're growing by one item, no?

Comment (peterl-stripe) [reply src/main/scala/co/teapot/tempest/graph/MemMappedDynamicDirectedGraphConverter.scala]: Right. But internally `addAll` does capacity doubling, while `size` doesn't (in Intellij I clicked through to the source code of the IntArrayList library).

Comment (gkk-stripe) [reply src/main/scala/co/teapot/tempest/graph/MemMappedDynamicDirectedGraphConverter.scala]: Oh, I get it. Could you add a line of comment saying that we rely on doubling growth implemented by `addAll`?

**Comment by gkk-stripe** [comment]
@peterl-stripe ping about the review feedback

**Comment by peterl-stripe** [comment]
Thanks for the ping!  I was thinking about several merchant modeling issues, and this got buried in my todo list. Go ahead and merge if this is ready, or let me know if I should make other changes.

**Comment by gkk-stripe** [comment]
LGTM
Do we need to push a new docker image for this?

**Pull request merged**